ts-client/src/vault/strategy/frakt.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  AccountMeta,
+  Connection,
+  PublicKey,
+  SYSVAR_CLOCK_PUBKEY,
+  SystemProgram,
+  Transaction,
+  TransactionInstruction,
+} from '@solana/web3.js';
+import { TokenInfo, loans } from '@mercurial-finance/frakt-sdk';
+import { TOKEN_PROGRAM_ID } from '@solana/spl-token';
+import BN from 'bn.js';
+
+import { SEEDS, SOL_MINT } from '../constants';
+import { AffiliateVaultProgram, VaultProgram } from '../types';
+import { getOrCreateATAInstruction } from '../utils';
+import { Strategy, StrategyHandler } from '.';
+
+const FRAKT_PROGRAM_ID = new PublicKey('A66HabVL3DzNzeJgcHYtRRNW1ZRMKwBfrdSR4kLsZ9DJ');
+const FRAKT_ADMIN_FEE_PUBKEY = new PublicKey('9aTtUqAnuSMndCpjcPosRNf3fCkrTQAV8C8GERf3tZi3');
+
+export default class FraktHandler implements StrategyHandler {
+  private connection: Connection;
+
+  constructor(program: VaultProgram) {
+    this.connection = program.provider.connection;
+  }
+
+  async withdraw(
+    tokenInfo: TokenInfo,
+    walletPubKey: PublicKey,
+    program: VaultProgram,
+    strategy: Strategy,
+    vault: PublicKey,
+    tokenVault: PublicKey,
+    feeVault: PublicKey,
+    lpMint: PublicKey,
+    userToken: PublicKey,
+    userLp: PublicKey,
+    amount: BN,
+    preInstructions: TransactionInstruction[],
+    postInstructions: TransactionInstruction[],
+    opt?: {
+      affiliate?: {
+        affiliateId: PublicKey;
+        affiliateProgram: AffiliateVaultProgram;
+        partner: PublicKey;
+        user: PublicKey;
+      };
+    },
+  ): Promise<Transaction> {
+    if (!walletPubKey) throw new Error('No user wallet public key');
+
+    const { timeBasedLiquidityPools, priceBasedLiquidityPools } = await loans.getAllProgramAccounts(
```

Comment:
hmm, can we find other way to decide whether which pool is timeBasedLiquidityPools or priceBasedLiquidityPools instead of fetching all pools?

Ex: In rust, we can specify by using anchor discriminator